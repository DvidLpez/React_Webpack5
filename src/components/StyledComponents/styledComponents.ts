import { Link } from "react-router-dom";
import styled from "styled-components";

export const Wrapper = styled.div`
  text-align: left;
  margin: ${(props) => props.theme.spaces.none} auto;
  max-width: 1500px;
  padding: ${(props) => props.theme.spaces.s5};

  @media screen and (min-width: 768px) {
    padding: ${(props) => props.theme.spaces.s30};
  }
`;

export const Title = styled.h1`
  font-size: ${(props) => props.theme.fonts.extraBig};
`;

export const Subtitle = styled.h2`
  font-size: ${(props) => props.theme.fonts.big};
  margin-bottom: ${(props) => props.theme.spaces.s50};
`;

export const TitleSection = styled.h3`
  font-size: ${(props) => props.theme.fonts.large};
  margin-top: ${(props) => props.theme.spaces.s50};
  margin-bottom: ${(props) => props.theme.spaces.s30};
`;


export const Text = styled.p`
  font-size: ${(props) => props.theme.fonts.default};
`;

export const Span = styled.span`
  font-size: ${(props) => props.theme.fonts.default};
  color: ${(props) => props.theme.colors.lightgrey};
`;

export const LinkHome = styled(Link)`
  color: ${(props) => props.theme.colors.title};
  background-color: ${(props) => props.theme.colors.grey};
  border-radius: ${(props) => props.theme.spaces.s5};
  margin-top: ${(props) => props.theme.spaces.s10};
  padding: ${(props) => props.theme.spaces.button};
  text-decoration: none;
  &:hover {
    -webkit-box-shadow: ${(props) => props.theme.colors.shadow};
    -moz-box-shadow: ${(props) => props.theme.colors.shadow};
    box-shadow: ${(props) => props.theme.colors.shadow};
  }
`;
