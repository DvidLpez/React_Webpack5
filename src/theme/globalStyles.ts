
import { createGlobalStyle } from "styled-components";
import { theme } from "./theme";


export const GlobalStyle = createGlobalStyle`

   html {
      box-sizing: border-box;
   }
   body {
         background-color: ${theme.colors.tertiary};
         font-family: ${theme.fonts.style};
         padding: ${theme.spaces.none};
         margin: ${theme.spaces.none};
   }
   *,
   ::after,
   ::before {
      box-sizing: inherit;
   }

   a{
      text-decoration: none;
      color: ${theme.colors.primary};
   }

   hr{
      border: none;
      border-bottom: solid 1px ${theme.colors.secondary};
   }

   .icon_brand {
      font-size: ${theme.fonts.big};
      margin-right: ${theme.spaces.s15};
      color: ${theme.colors.title};
   }
   .icon_sweat {
      font-size: ${theme.fonts.extra};
   }

   .icon_error {
         font-size: ${theme.fonts.extraBig};
      }

   @media screen and (min-width: 768px) {
   
      .icon_brand {
         font-size: ${theme.fonts.extra};
      }
      
   }
`;
