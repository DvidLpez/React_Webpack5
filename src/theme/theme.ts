import { DefaultTheme } from "styled-components";

export const theme: DefaultTheme = {
  fonts: {
    style: "'Source Sans Pro', sans-serif",
    small: "0.9em",
    default: "1em",
    medium: "1.2em",
    large: "1.6em",
    big: "2em",
    extra: "3em",
    extraBig: "5em",
    bold: "600",
  },
  colors: {
    primary: "#000",
    secondary: "#dbdbdb",
    tertiary: "#fafafa",
    grey: "#414141",
    lightgrey: "#7c7c7c",
    gradient:
      "linear-gradient(to right, #3e9e7e 0%, #2e755d 10%, #205040 20%, #414141 30%)",
    title: "#fff",
    success: "#008000",
    closed: "#c60404",
    danger: "#932323",
    backComment: "#fff9d6",
    shadow: "2px 4px 11px 1px rgba(138, 138, 138, 1)",
  },
  spaces: {
    none: "0",
    center: "center",
    s5: "5px",
    s10: "10px",
    s15: "15px",
    s20: "20px",
    s25: "25px",
    s30: "30px",
    s50: "50px",
    p55: "5px 5px",
    p100: "10px 0",
    p1520: "15px 20px",
    p1010: "10px 10px",
    p1515: "15px 15px",
    p2020: "20px 20px",
    p2525: "25px 25px",
    p3030: "30px 30px",
    button: "10px 40px",
    borderRadius: "5px",
  }
};
