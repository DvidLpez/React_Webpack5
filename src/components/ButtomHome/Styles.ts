import styled from "styled-components";
import { Wrapper } from "../../components/StyledComponents/styledComponents";

export const WrapperLink = styled(Wrapper)`
  padding-top: ${(props) => props.theme.spaces.s20};
  text-align: center;
`;


