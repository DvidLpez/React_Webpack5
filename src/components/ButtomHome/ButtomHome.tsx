import React, { FC, ReactElement } from "react";
import { LinkHome } from "../StyledComponents/styledComponents";
import { WrapperLink } from "./Styles";


const ButtomHome: FC = (): ReactElement => {

   return (
      <WrapperLink>
         <LinkHome  to="/">
            Go Home
         </LinkHome>
      </WrapperLink> 
   )
}
export default ButtomHome;
