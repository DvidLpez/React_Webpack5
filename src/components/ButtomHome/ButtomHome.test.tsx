import React from "react";
import ButtomHome from "./ButtomHome";
import { mount, shallow, render} from 'enzyme';
import { LinkHome } from "../StyledComponents/styledComponents";
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme/theme";
import { MemoryRouter } from "react-router-dom";

// export const mountWithTheme = (children: any) =>
//    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>);

export const renderWithTheme = (children: any) =>
   render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);

// export const shallowWithTheme = (children: any) =>
//    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>);


describe('TESTING HOME BUTTOM', () => {

   it(' *** Check url and Value in buttom home ***', () => {

      const wrapper = renderWithTheme(
         <MemoryRouter>
            <ButtomHome />
         </MemoryRouter>
      );

      let anchor = wrapper.find('a'),
      url = anchor.attr('href'),
      value = anchor.html();

      expect(url).toEqual('/');   // OK
      expect(value).toMatch('Go Home');  // OK
      expect(wrapper).toMatchSnapshot();  // OK
   });

});
