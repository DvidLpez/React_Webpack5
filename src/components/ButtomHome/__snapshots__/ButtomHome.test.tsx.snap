// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING HOME BUTTOM  *** Check url and Value in buttom home *** 1`] = `
<div
  class="sc-bdfBwQ sc-fubCfw kUdDos fwLbUW"
>
  <a
    class="sc-gKsewC fMfLPa"
    href="/"
  >
    Go Home
  </a>
</div>
`;
